app/actions/client.py

```diff
@@ -83,12 +84,19 @@ def validate_vehicles(cls, val):
 class PullObservationsTransmissionsResponse(pydantic.BaseModel):
     transmissions: List[TransmissionsResponse]
 
-    @pydantic.validator("transmissions", pre=True)
-    def validate_transmissions(cls, val):
+    @pydantic.validator("transmissions")
+    def validate_and_filter_transmissions(cls, val):
         if isinstance(val, list):
-            return val
+            result = {
+                "transmissions": reduce(
+                    lambda acc, item: {**acc, item.collar_serial_num: item.gmt_offset},
+                    val,
+                    {}
+                )
+            }
+            return result["transmissions"]
         # val is not a valid list, return an empty list instead
```

Comment:
I think this reduction should not be within a validate method.  I recommend doing that reduction in a separate function where you'll be able to handle errors and edge cases.
You might also want to use `dict.setdefault()` on your accumulator dictionary rather than creating a new dict for each record.